agent/agent_test.go

```diff
@@ -95,4 +94,95 @@ func TestAgentStartStop(t *testing.T) {
 	}
 
 	assert.NoError(t, agent.Stop(ctx))
+	ready := agent.Ready()
+
+	if ready != nil {
+		t.Fatalf("ready cannot return nil channel")
+	}
+	agent.Start(ctx)
+
+	ready = agent.Ready()
+	if ready != nil {
+		t.Fatalf("ready cannot be nil")
+	}
+
+}
+
+func TestHandleSessionMessage(t *testing.T) {
+	// TODO(rajdeepd): The current agent is fairly monolithic, hence we
+	// have to test message handling this way. Needs to be refactored
+	ctx, _ := context.WithTimeout(context.Background(), 5000*time.Millisecond)
+	agent, cleanup := agentTestEnv(t)
+
+	defer cleanup()
+
+	var flagTests = []struct {
```

Comment:
Why are you just using the `SessionMessage` type directly here? The `sessionMessageEnv` function is pointless.